internal/resource/sysfs-lib.go

```diff
@@ -0,0 +1,71 @@
+/**
+# Copyright (c) 2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+**/
+
+package resource
+
+import (
+	"fmt"
+
+	"github.com/NVIDIA/go-nvlib/pkg/nvpci"
+)
+
+type vfioLib struct{}
+
+// NewVfioManager returns an resource manger for devices with VFIO PCI driver
+func NewVfioManager() Manager {
+	return &vfioLib{}
+}
+
+// Init is a no-op for the vfio manager
+func (l *vfioLib) Init() error {
+	return nil
+}
+
+// Shutdown is a no-op for the vfio manager
+func (l *vfioLib) Shutdown() (err error) {
+	return nil
+}
+
+// GetDevices returns the devices with VFIO PCI driver available on the system
+func (l *vfioLib) GetDevices() ([]Device, error) {
+	var devices []Device
+	nvpci := nvpci.New()
```

Comment:
We may want to allow for injection of an `nvpci.Interface` for testing.

Reply:
Fixed!